app/controllers/schools/participants_controller.rb

```diff
@@ -19,6 +19,31 @@ def show
     @mentor = @participant.early_career_teacher_profile&.mentor
   end
 
+  def edit_name; end
+
+  def update_name
+    @participant.assign_attributes(params.require(:user).permit(:full_name))
+    render "schools/participants/edit_name" and return unless @participant.valid?
+
+    @participant.save!
+    set_success_message(heading: "The participant's name has been updated")
+    redirect_to schools_participant_path(id: @participant)
```

Comment:
```
if participant.update(params.require(:user).permit(:full_name))
  set_success_message(heading: "The participant's name has been updated")
  redirect_to schools_participant_path(id: @participant)
else
  render "schools/participants/edit_name"
end
```
seems to be a more 'railsy' way of doing this, I will let you choose whether you like it more or not :)